### ISSUE TITLE
browsermob proxy instance not starting every now and then because port is already taken

### ISSUE THREAD

**Issue by manuelkiessling** [journeymonitor/monitor#16]

**Comment by manuelkiessling** [comment]
Looks like the locking mechanism doesn't work reliably enough if a lot of monitor tasks start in parallel.

**Comment by manuelkiessling** [comment]
Needs to be cleanly solved via #14.

**Comment by manuelkiessling** [comment]
Actually, it looks like a lot more ports are claimed by browsermob process than there are testcases:

```
tcp6       0      0 :::35972                :::*                    LISTEN      0          28966217    22395/java
tcp6       0      0 :::38628                :::*                    LISTEN      0          28963307    22395/java
tcp6       0      0 :::36580                :::*                    LISTEN      0          28962212    22395/java
tcp6       0      0 :::45220                :::*                    LISTEN      0          28959436    22395/java
tcp6       0      0 :::54501                :::*                    LISTEN      0          28963648    22395/java
tcp6       0      0 :::34725                :::*                    LISTEN      0          28964025    22395/java
tcp6       0      0 :::42533                :::*                    LISTEN      0          28961224    22395/java
tcp6       0      0 :::39045                :::*                    LISTEN      0          28960494    22395/java
tcp6       0      0 :::53829                :::*                    LISTEN      0          28959423    22395/java
tcp6       0      0 :::54918                :::*                    LISTEN      0          28966488    22395/java
tcp6       0      0 :::52454                :::*                    LISTEN      0          28962664    22395/java
tcp6       0      0 :::35750                :::*                    LISTEN      0          28964869    22395/java
tcp6       0      0 :::55686                :::*                    LISTEN      0          28960744    22395/java
tcp6       0      0 :::59782                :::*                    LISTEN      0          28962289    22395/java
tcp6       0      0 :::44870                :::*                    LISTEN      0          28960077    22395/java
tcp6       0      0 :::33383                :::*                    LISTEN      0          28967454    22395/java
tcp6       0      0 :::54407                :::*                    LISTEN      0          28963766    22395/java
tcp6       0      0 :::60679                :::*                    LISTEN      0          28965072    22395/java
tcp6       0      0 :::42024                :::*                    LISTEN      0          28968135    22395/java
tcp6       0      0 :::49704                :::*                    LISTEN      0          28969012    22395/java
tcp6       0      0 :::52264                :::*                    LISTEN      0          28962705    22395/java
tcp6       0      0 :::55144                :::*                    LISTEN      0          28962581    22395/java
tcp6       0      0 :::58440                :::*                    LISTEN      0          28959718    22395/java
tcp6       0      0 :::36392                :::*                    LISTEN      0          28961409    22395/java
tcp6       0      0 :::54920                :::*                    LISTEN      0          28962058    22395/java
tcp6       0      0 :::14760                :::*                    LISTEN      0          28955365    22395/java
tcp6       0      0 :::50953                :::*                    LISTEN      0          28968114    22395/java
tcp6       0      0 :::55753                :::*                    LISTEN      0          28968063    22395/java
tcp6       0      0 :::33769                :::*                    LISTEN      0          28965697    22395/java
tcp6       0      0 :::32841                :::*                    LISTEN      0          28963711    22395/java
tcp6       0      0 :::40553                :::*                    LISTEN      0          28966502    22395/java
tcp6       0      0 :::41449                :::*                    LISTEN      0          28966067    22395/java
tcp6       0      0 :::35561                :::*                    LISTEN      0          28963955    22395/java
tcp6       0      0 :::36809                :::*                    LISTEN      0          28959508    22395/java
tcp6       0      0 :::46506                :::*                    LISTEN      0          28968103    22395/java
tcp6       0      0 :::53802                :::*                    LISTEN      0          28968048    22395/java
tcp6       0      0 :::48106                :::*                    LISTEN      0          28962413    22395/java
tcp6       0      0 :::54186                :::*                    LISTEN      0          28962273    22395/java
tcp6       0      0 :::33034                :::*                    LISTEN      0          28956619    22395/java
tcp6       0      0 :::40490                :::*                    LISTEN      0          4597167     22395/java
tcp6       0      0 :::45643                :::*                    LISTEN      0          28968105    22395/java
tcp6       0      0 :::48011                :::*                    LISTEN      0          28968061    22395/java
tcp6       0      0 :::40203                :::*                    LISTEN      0          28967425    22395/java
tcp6       0      0 :::58955                :::*                    LISTEN      0          28963556    22395/java
tcp6       0      0 :::45451                :::*                    LISTEN      0          28965218    22395/java
tcp6       0      0 :::57067                :::*                    LISTEN      0          28961745    22395/java
tcp6       0      0 :::37739                :::*                    LISTEN      0          28962115    22395/java
tcp6       0      0 :::56331                :::*                    LISTEN      0          28960362    22395/java
tcp6       0      0 :::50827                :::*                    LISTEN      0          28961854    22395/java
tcp6       0      0 :::36620                :::*                    LISTEN      0          28963491    22395/java
tcp6       0      0 :::56236                :::*                    LISTEN      0          28965156    22395/java
tcp6       0      0 :::48940                :::*                    LISTEN      0          28963941    22395/java
tcp6       0      0 :::40908                :::*                    LISTEN      0          28962120    22395/java
tcp6       0      0 :::56557                :::*                    LISTEN      0          28968120    22395/java
tcp6       0      0 :::52973                :::*                    LISTEN      0          28963580    22395/java
tcp6       0      0 :::48845                :::*                    LISTEN      0          28965244    22395/java
tcp6       0      0 :::45166                :::*                    LISTEN      0          28966376    22395/java
tcp6       0      0 :::36494                :::*                    LISTEN      0          28963567    22395/java
tcp6       0      0 :::46254                :::*                    LISTEN      0          28963324    22395/java
tcp6       0      0 :::57614                :::*                    LISTEN      0          28962297    22395/java
tcp6       0      0 :::50286                :::*                    LISTEN      0          28962062    22395/java
tcp6       0      0 :::57166                :::*                    LISTEN      0          28962048    22395/java
tcp6       0      0 :::37998                :::*                    LISTEN      0          28959392    22395/java
tcp6       0      0 :::59119                :::*                    LISTEN      0          28966866    22395/java
tcp6       0      0 :::48559                :::*                    LISTEN      0          28967427    22395/java
tcp6       0      0 :::48527                :::*                    LISTEN      0          28966355    22395/java
tcp6       0      0 :::40304                :::*                    LISTEN      0          28969054    22395/java
tcp6       0      0 :::50320                :::*                    LISTEN      0          28963658    22395/java
tcp6       0      0 :::52080                :::*                    LISTEN      0          28966364    22395/java
tcp6       0      0 :::52944                :::*                    LISTEN      0          28961660    22395/java
tcp6       0      0 :::39888                :::*                    LISTEN      0          28961013    22395/java
tcp6       0      0 :::60689                :::*                    LISTEN      0          28968977    22395/java
tcp6       0      0 :::37137                :::*                    LISTEN      0          28965313    22395/java
tcp6       0      0 :::46193                :::*                    LISTEN      0          28963310    22395/java
tcp6       0      0 :::45233                :::*                    LISTEN      0          28962690    22395/java
tcp6       0      0 :::41425                :::*                    LISTEN      0          28961414    22395/java
tcp6       0      0 :::50770                :::*                    LISTEN      0          28968154    22395/java
tcp6       0      0 :::34450                :::*                    LISTEN      0          28969026    22395/java
tcp6       0      0 :::38194                :::*                    LISTEN      0          28963764    22395/java
tcp6       0      0 :::60658                :::*                    LISTEN      0          28961726    22395/java
tcp6       0      0 :::44018                :::*                    LISTEN      0          28960685    22395/java
tcp6       0      0 :::48306                :::*                    LISTEN      0          28962146    22395/java
tcp6       0      0 :::57362                :::*                    LISTEN      0          28960057    22395/java
tcp6       0      0 :::42067                :::*                    LISTEN      0          28966714    22395/java
tcp6       0      0 :::34259                :::*                    LISTEN      0          28963670    22395/java
tcp6       0      0 :::36979                :::*                    LISTEN      0          28963662    22395/java
tcp6       0      0 :::42963                :::*                    LISTEN      0          28966259    22395/java
tcp6       0      0 :::44723                :::*                    LISTEN      0          28964528    22395/java
tcp6       0      0 :::39027                :::*                    LISTEN      0          28966100    22395/java
tcp6       0      0 :::33459                :::*                    LISTEN      0          28961783    22395/java
tcp6       0      0 :::51699                :::*                    LISTEN      0          28961722    22395/java
tcp6       0      0 :::43892                :::*                    LISTEN      0          28962707    22395/java
tcp6       0      0 :::32788                :::*                    LISTEN      0          28961469    22395/java
tcp6       0      0 :::38516                :::*                    LISTEN      0          28959582    22395/java
tcp6       0      0 :::38228                :::*                    LISTEN      0          28962078    22395/java
tcp6       0      0 :::52789                :::*                    LISTEN      0          28966693    22395/java
tcp6       0      0 :::44725                :::*                    LISTEN      0          28966528    22395/java
tcp6       0      0 :::59445                :::*                    LISTEN      0          28962699    22395/java
tcp6       0      0 :::42837                :::*                    LISTEN      0          28963957    22395/java
tcp6       0      0 :::40630                :::*                    LISTEN      0          28960693    22395/java
tcp6       0      0 :::51254                :::*                    LISTEN      0          28960224    22395/java
tcp6       0      0 :::32054                :::*                    LISTEN      0          28956068    22395/java
tcp6       0      0 :::32951                :::*                    LISTEN      0          28965169    22395/java
tcp6       0      0 :::38583                :::*                    LISTEN      0          28963195    22395/java
tcp6       0      0 :::42231                :::*                    LISTEN      0          28961705    22395/java
tcp6       0      0 :::59319                :::*                    LISTEN      0          28960708    22395/java
tcp6       0      0 :::48119                :::*                    LISTEN      0          28960803    22395/java
tcp6       0      0 :::47704                :::*                    LISTEN      0          28963735    22395/java
tcp6       0      0 :::40440                :::*                    LISTEN      0          28963694    22395/java
tcp6       0      0 :::33944                :::*                    LISTEN      0          28965663    22395/java
tcp6       0      0 :::59288                :::*                    LISTEN      0          28963619    22395/java
tcp6       0      0 :::53208                :::*                    LISTEN      0          28965237    22395/java
tcp6       0      0 :::55800                :::*                    LISTEN      0          28966060    22395/java
tcp6       0      0 :::40344                :::*                    LISTEN      0          28962269    22395/java
tcp6       0      0 :::43224                :::*                    LISTEN      0          28960488    22395/java
tcp6       0      0 :::36568                :::*                    LISTEN      0          28961927    22395/java
tcp6       0      0 :::39481                :::*                    LISTEN      0          28964983    22395/java
tcp6       0      0 :::59513                :::*                    LISTEN      0          28961740    22395/java
tcp6       0      0 :::38873                :::*                    LISTEN      0          28960505    22395/java
tcp6       0      0 :::59098                :::*                    LISTEN      0          28961679    22395/java
tcp6       0      0 :::44186                :::*                    LISTEN      0          28962406    22395/java
tcp6       0      0 :::41467                :::*                    LISTEN      0          28966554    22395/java
tcp6       0      0 :::32827                :::*                    LISTEN      0          28966538    22395/java
tcp6       0      0 :::33723                :::*                    LISTEN      0          28965251    22395/java
tcp6       0      0 :::41531                :::*                    LISTEN      0          28959626    22395/java
tcp6       0      0 :::56059                :::*                    LISTEN      0          28959450    22395/java
tcp6       0      0 :::42780                :::*                    LISTEN      0          28966726    22395/java
tcp6       0      0 :::36412                :::*                    LISTEN      0          28968067    22395/java
tcp6       0      0 :::51260                :::*                    LISTEN      0          28964596    22395/java
tcp6       0      0 :::47548                :::*                    LISTEN      0          28962528    22395/java
tcp6       0      0 :::57404                :::*                    LISTEN      0          28960217    22395/java
tcp6       0      0 :::43773                :::*                    LISTEN      0          28966768    22395/java
tcp6       0      0 :::43709                :::*                    LISTEN      0          28967186    22395/java
tcp6       0      0 :::53117                :::*                    LISTEN      0          28963539    22395/java
tcp6       0      0 :::46333                :::*                    LISTEN      0          28965299    22395/java
tcp6       0      0 :::43645                :::*                    LISTEN      0          28962562    22395/java
tcp6       0      0 :::51102                :::*                    LISTEN      0          28966690    22395/java
tcp6       0      0 :::56478                :::*                    LISTEN      0          28961708    22395/java
tcp6       0      0 :::41503                :::*                    LISTEN      0          28965727    22395/java
tcp6       0      0 :::56031                :::*                    LISTEN      0          28962187    22395/java
tcp6       0      0 :::57984                :::*                    LISTEN      0          28967416    22395/java
tcp6       0      0 :::57952                :::*                    LISTEN      0          28963698    22395/java
tcp6       0      0 :::50176                :::*                    LISTEN      0          28965657    22395/java
tcp6       0      0 :::60256                :::*                    LISTEN      0          28966369    22395/java
tcp6       0      0 :::36192                :::*                    LISTEN      0          28966120    22395/java
tcp6       0      0 :::36864                :::*                    LISTEN      0          28965159    22395/java
tcp6       0      0 :::45568                :::*                    LISTEN      0          28961428    22395/java
tcp6       0      0 :::55360                :::*                    LISTEN      0          28962285    22395/java
tcp6       0      0 :::45760                :::*                    LISTEN      0          28960502    22395/java
tcp6       0      0 :::50432                :::*                    LISTEN      0          28960270    22395/java
tcp6       0      0 :::40928                :::*                    LISTEN      0          28956614    22395/java
tcp6       0      0 :::60673                :::*                    LISTEN      0          28966706    22395/java
tcp6       0      0 :::51617                :::*                    LISTEN      0          28965257    22395/java
tcp6       0      0 :::45377                :::*                    LISTEN      0          28961381    22395/java
tcp6       0      0 :::51522                :::*                    LISTEN      0          28966738    22395/java
tcp6       0      0 :::40738                :::*                    LISTEN      0          28963761    22395/java
tcp6       0      0 :::59522                :::*                    LISTEN      0          28963706    22395/java
tcp6       0      0 :::44546                :::*                    LISTEN      0          28966245    22395/java
tcp6       0      0 :::53890                :::*                    LISTEN      0          28962421    22395/java
tcp6       0      0 :::59298                :::*                    LISTEN      0          28960925    22395/java
tcp6       0      0 127.0.0.1:9090          :::*                    LISTEN      0          2528447     22395/java
tcp6       0      0 :::41571                :::*                    LISTEN      0          28963624    22395/java
tcp6       0      0 :::58627                :::*                    LISTEN      0          28963482    22395/java
tcp6       0      0 :::42115                :::*                    LISTEN      0          28964556    22395/java
tcp6       0      0 :::54307                :::*                    LISTEN      0          28966186    22395/java
tcp6       0      0 :::37027                :::*                    LISTEN      0          28963408    22395/java
tcp6       0      0 :::38115                :::*                    LISTEN      0          28963299    22395/java
tcp6       0      0 :::52227                :::*                    LISTEN      0          28962883    22395/java
tcp6       0      0 :::42947                :::*                    LISTEN      0          28960211    22395/java
```

(there are only 12 testcases, thus there should only be 1 master proxy process and 12 children processes)

**Comment by manuelkiessling** [comment]
Seems solved with 9b94701.
